scality_manila_utils/cli.py

```diff
@@ -98,109 +99,163 @@ def main(args=None):
+
+    def add_parser_get():
+        for (protocol, subparsers) in protocol_subparsers:
+            help = description = ('List the addresses that a filesystem is '
+                                  'exported to')
+            parser_get = subparsers.add_parser(
+                'get', description=description,
+                help=help
+            )
+            parser_get.add_argument(
+                'export_name',
+                help='Filesystem to get information about'
+            )
+            parser_get.set_defaults(protocol=protocol, func='get_export')
+
+    def add_parser_wipe():
+        for (protocol, subparsers) in protocol_subparsers:
+            help = description = ('Remove an existing export. The export '
+                                  'must not have any grants.')
+            parser_wipe = subparsers.add_parser(
+                'wipe', description=description,
+                help=help
+            )
+            parser_wipe.add_argument(
+                'export_name',
+                help='Filesystem to remove'
+            )
+            parser_wipe.set_defaults(protocol=protocol, func='wipe_export')
+
+    def add_parser_grant():
+        for (protocol, subparsers) in protocol_subparsers:
+            help = description = 'Grant access to an existing filesystem'
+
+            if protocol == "nfs":
+                help += "and reexport it"
+                description += "and reexport it"
+
+            parser_grant = subparsers.add_parser(
+                'grant', description=description, help=help,
+            )
+            parser_grant.add_argument(
+                'export_name',
+                help='Filesystem to grant access to'
+            )
+            parser_grant.add_argument(
+                'host',
+                help='IP address or network to grant access for'
+            )
+            parser_grant.set_defaults(protocol=protocol, func='grant_access')
+
+            if protocol == "nfs":
+                parser_grant.add_argument(
+                    'options',
+                    help='Export options',
+                    nargs='*'
+                )
+
+    def add_parse_revoke():
+        for (protocol, subparsers) in protocol_subparsers:
+            help = description = 'Revoke access from an existing filesystem'
+
+            if protocol == "nfs":
+                help += "and reexport it"
+                description += "and reexport it"
+
+            parser_revoke = subparsers.add_parser(
+                'revoke', description=description,
+                help=help
+            )
+            parser_revoke.add_argument(
+                'export_name',
+                help='Filesystem to revoke access from'
+            )
+            parser_revoke.add_argument(
+                'host',
+                help='IP address or network to revoke access for'
+            )
+
+            parser_revoke.set_defaults(protocol=protocol, func='revoke_access')
+
```

Comment:
Well I agree this is suboptimal. But imo it's not worth the change because I am not sure it will be way better. Sometime I prefer some sort of code duplication instead of a generic function with 7 arguments. Just a personal choice.